src/network/event_loop.rs

```diff
@@ -299,28 +271,60 @@ impl EventLoop {
 				match event {
 					IdentifyEvent::Received {
 						peer_id,
-						info: Info {
-							listen_addrs,
-							protocols,
-							..
-						},
+						info: Info { listen_addrs, .. },
 					} => {
-						debug!("Identify Received event. PeerId: {peer_id:?}. Listen address: {listen_addrs:?}");
-
-						if protocols
-							.iter()
-							.any(|p| p.as_bytes() == protocol::DEFAULT_PROTO_NAME)
+						info!("Identity Received from: {peer_id:?} on listen address: {listen_addrs:?}");
+						// before we try and do a reservation with the relay
+						// we have to exchange observed addresses
+						// in this case relay needs to tell us our own
+						if peer_id == self.relay_reservation.id
+							&& self.relay_reservation.is_reserved == false
 						{
-							for addr in listen_addrs {
-								self.swarm
-									.behaviour_mut()
-									.kademlia
-									.add_address(&peer_id, addr);
+							match self.swarm.listen_on(
+								self.relay_reservation
```

Comment:
What does the relay reservation actually do here?

Reply:
stops node from grabbing more reservations